test/workbox-routing/node/Router.mjs

```diff
@@ -0,0 +1,302 @@
+
+      expect(router.defaultHandler).to.eql({handle: HANDLER});
+    });
+
+    it(`should update the expected internal state when setCatchHandler() is called`, function() {
+      const router = new Router();
+      router.setCatchHandler(HANDLER);
+
+      expect(router.catchHandler).to.deep.eql({handle: HANDLER});
+    });
+  });
+
+  describe(`handling requests`, function() {
+    it(`should return a response from the Route's handler when there's a matching route`, async function() {
+      const router = new Router();
+      const route = new Route(
+        () => true,
+        () => new Response(EXPECTED_RESPONSE_BODY),
+      );
+      router.registerRoute(route);
+
+      // route.match() always returns true, so the Request details don't matter.
+      const event = new FetchEvent('fetch', {request: new Request(self.location)});
+      const response = await router.handleRequest(event);
+      const responseBody = await response.text();
+
+      expect(responseBody).to.eql(EXPECTED_RESPONSE_BODY);
+    });
+
+    it(`should return a response from the default handler when there's no matching route`, async function() {
+      const router = new Router();
+      const route = new Route(
+        () => false,
+        () => new Response(),
+      );
+      router.registerRoute(route);
+      router.setDefaultHandler(() => new Response(EXPECTED_RESPONSE_BODY));
+
+      // route.match() always returns false, so the Request details don't matter.
+      const event = new FetchEvent('fetch', {request: new Request(self.location)});
+      const response = await router.handleRequest(event);
+      const responseBody = await response.text();
+
+      expect(responseBody).to.eql(EXPECTED_RESPONSE_BODY);
+    });
+
+    it(`should return a response from the catch handler when the matching route's handler rejects`, async function() {
+      const router = new Router();
+      const route = new Route(
+        () => true,
+        () => Promise.reject(),
+      );
+      router.registerRoute(route);
+      router.setCatchHandler(() => new Response(EXPECTED_RESPONSE_BODY));
+
+      // route.match() always returns false, so the Request details don't matter.
+      const event = new FetchEvent('fetch', {request: new Request(self.location)});
+      const response = await router.handleRequest(event);
+      const responseBody = await response.text();
+
+      expect(responseBody).to.eql(EXPECTED_RESPONSE_BODY);
+    });
+
+    it(`should not return a response when there's no matching route and no default handler`, async function() {
+      const router = new Router();
+      const route = new Route(
+        () => false,
+        () => new Response(),
+      );
+      router.registerRoute(route);
+
+      // route.match() always returns false, so the Request details don't matter.
+      const event = new FetchEvent('fetch', {request: new Request(self.location)});
+      const response = await router.handleRequest(event);
```

Comment:
Here you are await'ing but I don't think you were in the fetch event listener.

Reply:
I could see how that would be confusing. Thanks for pointing that out.